### PR TITLE
dash/dg2-api-default-changes

### PR DESCRIPTION
**Pull request by ddragula**
- [x] changed `defaults.columns` to `columnDefaults`
- [x] extracted `caption` to root from `settings`
- [x] added `getOptionsJSON` method to the DataGrid

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208087887101609

### CODE REVIEW

File: ts/DataGrid/Content/TableCell.ts [hunk 1]
```diff
@@ -98,9 +98,11 @@ class TableCell extends Cell {
     /**
      * Renders the cell.
      */
-    public render(): void {
+    public override render(): void {
```
Comment (karolkolodziej) [inline ts/DataGrid/Content/TableCell.ts:101]: Is this intended?

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
1. Please check failing cypress tests. 
2. Update paths in DataGridComponent editableOptions

**Review by stitot — APPROVED** [review]

File: ts/DataGrid/Content/TableCell.ts (hunk 1 above)
Comment (ddragula) [reply ts/DataGrid/Content/TableCell.ts:101]: yes

**Review by karolkolodziej — APPROVED** [review]

File: docs/datagrid/understanding-datagrid.md [hunk 2]
```diff
@@ -10,29 +10,25 @@ Defaults
...
-    caption: {
-        text: 'Your title of datagrid'Heade
-    }
+caption: {
+    text: 'Title of the DataGrid'
 }
 ```
 
-For more information on `caption` options see the [API reference]().
+For more information on `caption` options see the [API reference](http://localhost:9005/dashboards/#interfaces/DataGrid_Options.Options-1#caption).
```
Comment (sebastianbochan) [inline docs/datagrid/understanding-datagrid.md:31]: It should not be a reference to localhost.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

File: docs/datagrid/understanding-datagrid.md (hunk 2 above)
Comment (ddragula) [reply docs/datagrid/understanding-datagrid.md:31]: oops, good catch :D

Comment (ddragula) [inline docs/datagrid/understanding-datagrid.md:31]:
```suggestion
For more information on `caption` options see the [API reference](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.Options-1#caption).
```

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**